src/render.py

```diff
@@ -454,28 +447,22 @@ async def get_post_urls_and_title(
 
 def get_images_as_epub_items(image_map: ImageMap):
     items = []
-    for url, icon in image_map.icons.items():
-        filename = image_map.get_icon_name(url)
-        if filename is None:
-            continue
-        items.append(
-            epub.EpubItem(
-                uid=filename,
-                file_name=filename,
-                media_type=icon.media_type,
-                content=icon.file,
-            )
-        )
-    for url, image in image_map.images.items():
-        filename = image_map.get_image_name(url)
+    for url, mapped_image in image_map.map.items():
+        match mapped_image.name:
+            case "icon":
+                filename = image_map.get_icon_name(url)
+            case "image":
+                filename = image_map.get_image_name(url)
```

Comment:
I think you could merge `get_icon_name` and `get_image_name` together, and move this switch inside them (since it has access to the mapped image).

I don't feel strongly about this though. LMK when you want me to merge this.

Reply:
I prefer the current setup in that regard, I think. On my end I'm pretty sure it's ready to merge! No hanging bugs I've found that I'm working on fixes for, or anything.